modules/extensions/src/main/java/org/apache/synapse/mediators/bsf/ScriptMediator.java

```diff
@@ -593,7 +594,14 @@ protected void initScriptEngine() {
         if (language.equals(NASHORN_JAVA_SCRIPT)) {
             this.scriptEngine = engineManager.getEngineByName(NASHORN);
         } else {
-            this.scriptEngine = engineManager.getEngineByExtension(language);
+            try {
+                this.scriptEngine = engineManager.getEngineByExtension(language);
+            } finally {
+                if (scriptEngine == null ) {
```

Comment:
We need to do this because getEngineByExtension method of bsf library haven't handled the exception so the execution directly goes to the finally method of AbstractSynapseArtifactDeployer's deploy method without deploying other artifacts and that was the exact scenario which this fix tries to avoid.